### PR TITLE
Add Open Graph metadata to blog post to improve social share

### LINKED ISSUE #15: Add Open Graph metadata to blog post to improve social share snippet
https://developers.facebook.com/docs/sharing/webmasters

### PR DESCRIPTION
**Pull request by lumie31**
#### What does this PR do?
- Add Open Graph metadata to blog post to improve social share
#### Description of Task to be completed?
- In order to show correct snippet in social shares we want to use Open Graph metatags in Post show:
```
<meta property="og:title" content="Post title" />
<meta property="og:description" content="The first characters of Post" />
<meta property="og:image" content="URL of first image in Post" />
```
#### Any background context you want to provide?
- Most content is shared to Facebook as a URL, so it's important that you mark up your website with Open Graph tags to take control over how your content appears on Facebook 

@sseerrggii  This PR aims to fix #15, I have doubts with the values in the `content` attribute which is partially because I couldn't test this out because of the limitation of my pc and also I felt the content are supposed to be dynamic/changing with different post as opposed to them being static. Please review and give feedback

### CODE REVIEW

**Comment by sseerrggii** [comment]
Hi @lumie31 It works! but we should to define cover-photo and description in metadata at top of the post markdown document.

I would request one improvement. If title, description or cover-foto is empty is better don't show this metatag line. For instance if { page.description }} is empty we don't want to show the line: 

`<meta property="og:description" content="{{ page.description }}" />`

**Comment by lumie31** [comment]
@sseerrggii Can I get more clarity into what you want to be done? I don't seem to totally understand what you said above

**Comment by sauloperez** [comment]
What we want is to conditionally populate each of these meta properties depending on the post's metadata listed in the [frontmatter](https://jekyllrb.com/docs/front-matter/). So if a particular blog post doesn't have `description` specified we shouldn't render that meta tag (see an example in https://github.com/coopdevs/coopdevs.github.io/pull/61). This way we won't show empty fields when the open graph is rendered.

You should be able to do that using [Liquid](https://shopify.github.io/liquid/tags/control-flow/), the templating language Jekyll uses.

Lastly, I recommend you extract this change into its own HTML in [_includes/](https://github.com/coopdevs/coopdevs.github.io/tree/develop/_includes) as we did with other components of the site to keep things isolated.

**Comment by lumie31** [comment]
Hello, I made some changes based on the above suggestions. Kindly review and let me know if anything else needs to be done. Cheers!!!

File: _includes/meta.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{% if page.title %}
+<meta property="og:title" content="{{ page.title }}" />
+{% endif %} {% if page.description %}
```
Comment (sauloperez) [inline _includes/meta.html]: I would rather move the `if` statement to a separate line. It's what we do everywhere else.

EDIT sorry, I wrote the comment yesterday but I didn't submit the review. It's not the first time that happens to me :see_no_evil:

**Comment by sseerrggii** [comment]
Works fine for me! :+1: 

Tanks @lumie31

**Review by sseerrggii — APPROVED** [review]

File: _includes/meta.html (hunk 1 above)
Comment (lumie31) [reply _includes/meta.html]: @sauloperez I actually wrote it on a separate line but my text editor kept formatting it this way. I'd just disable `format-on-save` and make the push.

About your comment, I guess that can happen to anyone. It happens to me too when chatting, I send messages in my mind without remembering to press the `Enter` key. Haha

File: _includes/meta.html [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+{% endif %} 
+
+{% if page.description %}
 <meta property="og:description" content="{{ page.description }}" />
-{% endif %} {% if page.cover-photo %}
+{% endif %} 
+
+{% if page.cover-photo %}
 <meta property="og:image" content="{{ page.cover - photo }}" />
 {% endif %}
```
Comment (lumie31) [inline _includes/meta.html]: @sseerrggii @sauloperez Fixed the indentation in the file. Cheers!

File: _includes/meta.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{% if page.title %}
+<meta property="og:title" content="{{ page.title }}" />
+{% endif %} 
+
+{% if page.description %}
+<meta property="og:description" content="{{ page.description }}" />
+{% endif %} 
+
+{% if page.cover-photo %}
+<meta property="og:image" content="{{ page.cover - photo }}" />
```
Comment (sauloperez) [inline _includes/meta.html]: ups, sorry, there is a typo here. It should be `page.cover-photo`

Comment (lumie31) [reply _includes/meta.html]: @sauloperez  Fixed the typo. Thanks for pointing out

**Review by sauloperez — APPROVED** [review]
Awesome!

**Pull request merged**